### PR TITLE
Enable Service charts edit

### PR DESCRIPTION
**Pull request by torchiaf**
### Summary
Fixes https://github.com/epinio/ui/issues/339

### CODE REVIEW

File: dashboard/pkg/epinio/edit/services.vue [hunk 1]
```diff
@@ -134,6 +134,13 @@ export default Vue.extend<Data, EpinioCompRecord, EpinioCompRecord, EpinioCompRe
   methods: {
     async save(saveCb: (success: boolean) => void) {
       this.errors = [];
+
+      const newSettings = !isEqual(objValuesToString(this.chartValues), objValuesToString(this.value.settings));
```
Comment (richard-cox) [inline dashboard/pkg/epinio/edit/services.vue:138]: `objValuesToString(this.chartValues)` is called twice. This can be done just once above `const chartValues = objValuesToString(this.chartValues)`

**Review by richard-cox — APPROVED** [review]
One comment, but I think we can ignore that and get this merged

**Comment by thehejik** [comment]
TL;DR
There are two problems:
* `Save` button is always active in `Edit Service` even when the original values were not touched yet
* (backend related?) Sometimes the service is not modified accordingly and doesn't use the new settings/values, the pods are not restarted.

This is the epinio-server log message after unsuccessful service edit:
```
2023/11/15 15:03:06 EpinioServer: "msg"="installing or upgrading service ASYNC" "error"="cannot patch \"xf448270cb25596849f7da68cc872-mysql\" with kind StatefulSet: StatefulSet.apps \"xf448270cb25596849f7da68cc872-mysql\" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'ordinals', 'template', 'updateStrategy', 'persistentVolumeClaimRetentionPolicy' and 'minReadySeconds' are forbidden" "requestId"="d7cc5971-c085-484d-9da4-062327a1e797"
```

Test report:
Now I'm able to Edit deployed service with settings over UI but the resources are often not update at all - maybe it depends on the used chart?

This is how I tested:
* Modify existing service `mysql-dev` and provide compatible settings:
```bash
$ k edit services.application.epinio.io -n epinio mysql-dev
  # put this at the values level
  settings:                                                
    image.pullSecret:
      type: string                         
    auth.rootPassword:                     
      type: string                                                       
    global.storageClass:      
      enum:           
      - default       
      - local-path      
      - none            
      type: string
    primary.persistence.enabled:
      type: bool                           
    primary.service.ports.mysql:
      maximum: "3307"           
      minimum: "3306"           
      type: integer
```
* Deploy the service over UI - the service is deployed successfully - so far only basic data types: int,bool,string are supported - no array or maps yet:
image
* Click three dots and perform `Edit Service`. The Settings fields are active now and values could be modified (validation of the int field is ok):
image
* but very often the service settings are actually not updated for some reason

**Comment by andreas-kupries** [comment]
Did you replicate it with the CLI as well ? See `epinio service update`.

The message about `StatefulSet` from the server definitely is from deeper in the system, not E itself.
Kubernetes itself rejects the change to the resource we are attempting.

> maybe it depends on the used chart?

I strongly suspect so just from reading this.
IOW a chart using one or more StatefulSets will reject, others can be ok.
I have no immediate fix for this, nor ideas to one.
For charts with StatefulSets updating the service in flight might not be possible at all.
I.e. require complete deletion and recreation of the service with the new settings.

**Comment by thehejik** [comment]
Edit is working flawlessly with bitnami nginx service.

I guess the previous problem can be related to `values.extraDeploy` defined for 
incorporated Epinio service resources (SA, RoleBinding and Job related to PVC) which are out of control of the helm chart.

Custom nginx service definition:
```yaml
kubectl apply -f - <<EOF
apiVersion: application.epinio.io/v1
kind: Service
metadata:
  name: nginx-dev
  namespace: epinio
  annotations:                                                                                                                                                              
    meta.helm.sh/release-name: epinio                                                                                                                                            
    meta.helm.sh/release-namespace: epinio
spec:
  name: nginx-dev
  shortDescription: A nginx test
  description: A simple nginx
  chart: nginx
  chartVersion: 15.2.0
  serviceIcon: https://bitnami.com/assets/stacks/nginx/img/nginx-stack-220x234.png
  appVersion: 15.2.0
  helmRepo:
    name: bitnami
    url: https://charts.bitnami.com/bitnami
  settings:
    replicaCount:
      type: integer
    ingress.enabled:
      type: bool
    containerPorts.http:
      type: integer
  values: |-
    service:
      type: ClusterIP
EOF
```

**Comment by thehejik** [comment]
@andreas-kupries it works fine also with mysql service (eg. one can change `port` and `password`) but changing custom settings related to storage (`storageClass` and `persistence`) is not good idea as it likely clashes with `extraDeploy` resources out of service helm chart control.
